src/public/authentication.ts

```diff
@@ -0,0 +1,411 @@
+import { ensureInitialized, sendMessageRequest, waitForMessageQueue } from "../internal/internalAPIs";
+import { GlobalVars } from "../internal/globalVars";
+import { frameContexts } from "../internal/constants";
+import { HostClientType } from "./constants";
+/**
+ * Namespace to interact with the authentication-specific part of the SDK.
+ * This object is used for starting or completing authentication flows.
+ */
+export namespace authentication {
+  let authParams: AuthenticateParameters;
+  let authWindowMonitor: number;
+  GlobalVars.handlers["authentication.authenticate.success"] = handleSuccess;
+  GlobalVars.handlers["authentication.authenticate.failure"] = handleFailure;
+  /**
+   * Registers the authentication GlobalVars.handlers
+   * @param authenticateParameters A set of values that configure the authentication pop-up.
+   */
+  export function registerAuthenticationHandlers(authenticateParameters: AuthenticateParameters): void {
+    authParams = authenticateParameters;
+  }
+  /**
```

Comment:
NIT: Add an empty line between methods for some better visual separation and readability